packages/api-format/src/codec/sample.json

```diff
@@ -0,0 +1,2199 @@
+              "name": "propose",
+              "arguments": [
+                {
+                  "name": "origin",
+                  "type": "T::Origin"
+                },
+                {
+                  "name": "proposal",
+                  "type": "Box<T::Proposal>"
+                },
+                {
+                  "name": "value",
+                  "type": "T::Balance"
+                }
+              ],
+              "documentation": []
+            },
+            {
+              "id": "0x0001",
+              "name": "second",
+              "arguments": [
+                {
+                  "name": "origin",
+                  "type": "T::Origin"
+                },
+                {
+                  "name": "proposal",
+                  "type": "PropIndex"
+                }
+              ],
+              "documentation": []
+            },
+            {
+              "id": "0x0002",
+              "name": "vote",
+              "arguments": [
+                {
+                  "name": "origin",
+                  "type": "T::Origin"
+                },
+                {
+                  "name": "ref_index",
+                  "type": "ReferendumIndex"
+                },
+                {
+                  "name": "approve_proposal",
+                  "type": "bool"
+                }
+              ],
+              "documentation": []
+            },
+            {
+              "id": "0x0003",
+              "name": "start_referendum",
+              "arguments": [
+                {
+                  "name": "proposal",
+                  "type": "Box<T::Proposal>"
+                },
+                {
+                  "name": "vote_threshold",
+                  "type": "VoteThreshold"
+                }
+              ],
+              "documentation": []
+            },
+            {
+              "id": "0x0004",
+              "name": "cancel_referendum",
+              "arguments": [
+                {
+                  "name": "ref_index",
+                  "type": "ReferendumIndex"
+                }
+              ],
+              "documentation": []
+            }
+          ]
+        }
+      },
+      "storage": {
+        "prefix": "Democracy",
+        "functions": [
+          {
+            "name": "PublicPropCount",
+            "modifier": 1,
+            "type": "PropIndex",
+            "documentation": [
+              " The number of (public) proposals that have been made so far."
+            ]
+          },
+          {
+            "name": "PublicProps",
+            "modifier": 1,
+            "type": "Vec<(PropIndex, T::Proposal, T::AccountId)>",
+            "documentation": [
+              " The public proposals. Unsorted."
+            ]
+          },
+          {
+            "name": "DepositOf",
+            "modifier": 0,
+            "type": {
+              "key": "PropIndex",
+              "value": "(T::Balance, Vec<T::AccountId>)"
```

Comment:
This effective translates to -

```
{
  balance: Balance,
  vecAvvountId: Array<AccountId>
}
```

or rather as a CodecStruct -

```
extends class CodecStruct<{
  balance: Balance,
  vecAccountId: CodecArray<AccountId>
}> {
  constructor (value?: any) {
    super({
      balance: Balance,
      vecAccountId: CodecArray.with(AccountId)
    });
  }
}
```

And sadly, looking at the above it is pretty straight forward to generate, however... since we stll have the issue earlier with the getters not being automagically defined, it is a bit of an issue. Definition is one thing, but we really don't want people to go `instance.raw.balance.toNumber()` when they can do `instance.balance.toNumber()` - if no other way, maybe trim `raw` to something like `v`, i.e. shorter.

Dunno, just thinking through.